core/src/epicli/data/common/ansible/playbooks/roles/preflight/tasks/main.yml

```diff
@@ -2,14 +2,18 @@
 - include_tasks: check-user.yml
 
 - name: Set unsupported_components_checklist fact
+  delegate_to: localhost
+  become: false
   set_fact:
-     unsupported_components_checklist: "{{ unsupported_roles | selectattr('distro','equalto', ansible_distribution) | selectattr('arch','equalto', ansible_architecture) | map(attribute='roles') | list | flatten | intersect(group_names) }}"
+     unsupported_components_checklist: "{{ unsupported_roles | selectattr('distro','equalto', hostvars[groups['repository'][0]]['ansible_distribution']) | selectattr('arch','equalto', hostvars[groups['repository'][0]]['ansible_architecture']) | map(attribute='roles') | list | flatten | intersect(group_names) }}"
```

Comment:
IMO we should check all groups, not only `repository` since we have "custom repository" feature so it's possible to have no `repository` group.

More info here:
https://github.com/epiphany-platform/epiphany/blob/develop/docs/home/howto/CLUSTER.md#how-to-create-an-epiphany-cluster-using-custom-system-repository-and-docker-image-registry